addons/web_editor/static/src/js/wysiwyg/widgets/link_dialog.js

```diff
@@ -43,9 +43,10 @@ var LinkDialog = Dialog.extend({
 
         this.data = linkInfo || {};
         this.isButton = this.data.isButton;
+        const isButtonLink = this.isButton || (this.data.range.isOnAnchor() && this.data.className.includes("btn-link"));
```

Comment:
```suggestion
        const isButtonLink = this.isButton || (this.data.range && this.data.range.isOnAnchor() && this.data.className.includes("btn-link"));
```
below we are sing the range but testing if there is a range before, so it should still be done I think to be safe